src/contexts/FileContext.tsx

```diff
@@ -220,10 +220,23 @@ export const FileProvider: React.FC<{ children: ReactNode }> = ({ children }) =>
     if (filePath !== null && needToWriteEditorContentToDisk && _editor) {
       const markdownContent = getMarkdown(_editor)
       if (markdownContent !== null) {
+        const firstLine = markdownContent.split('\n')[0]
+        const newFileName = `${firstLine
```

Comment:
can this sanitization and file naming code be moved to a separate function please